Client/Frontend/Home/HomePanelViewController.swift

```diff
@@ -101,72 +121,45 @@ class HomePanelViewController: UIViewController, UITextFieldDelegate, HomePanelD
         }
     }
 
-    override func preferredStatusBarStyle() -> UIStatusBarStyle {
-        return UIStatusBarStyle.LightContent
-    }
-
     private func updateButtons() {
-        for index in 0...panels.count-1 {
-            let item = panels[index]
-            // If we have a button, we'll just reuse it
-            if (index < buttons.count) {
-                let button = buttons[index]
-                // TODO: Write a better equality check here.
-                if (item.title == button.item.title) {
-                    continue
-                }
-
-                button.item = item
-            } else {
-                // Otherwise create one
-                let toolbarButton = ToolbarButton(toolbarItem: item)
-                buttonContainerView.addSubview(toolbarButton)
-                toolbarButton.addTarget(self, action: "tappedButton:", forControlEvents: UIControlEvents.TouchUpInside)
-                buttons.append(toolbarButton)
-            }
-        }
-
-        // Now remove any extra buttons we find
-        // Note, since we modify index in the loop, we have to use the old for-loop syntax here.
-        // XXX - There's probably a better way to do this
-        for (var index = panels.count; index < buttons.count; index++) {
-            let button = buttons[index]
+        // Remove any existing buttons if we're rebuilding the toolbar.
+        for button in buttons {
             button.removeFromSuperview()
-            buttons.removeAtIndex(index)
-            index--
-        }
-    }
-
-    override func viewDidLoad() {
-        view.backgroundColor = BackgroundColor
-
-        buttonContainerView = ToolbarContainerView()
-        buttonContainerView.backgroundColor = BackgroundColor
-        view.addSubview(buttonContainerView)
-
-        controllerContainerView = UIView()
-        view.addSubview(controllerContainerView)
-
-        buttonContainerView.snp_makeConstraints { make in
-            make.top.left.right.equalTo(self.view)
-            make.height.equalTo(90)
         }
+        buttons.removeAll()
+
+        var prev: UIView? = nil
+        for panel in panels {
+            let button = UIButton()
+            buttonContainerView.addSubview(button)
+            button.addTarget(self, action: "SELtappedButton:", forControlEvents: UIControlEvents.TouchUpInside)
+            let image = UIImage(named: "nav-\(panel.imageName).png")!
+            button.setImage(getOverlayedImage(image, withColor: NormalIconColor), forState: UIControlState.Normal)
+            button.setImage(getOverlayedImage(image, withColor: SelectedIconColor), forState: UIControlState.Selected)
+            buttons.append(button)
```

Comment:
I have a patch that makes a generic Toolbar UIView that does some of this for us. We should switch over to something like that eventually.